app/models/spam_address.rb

```diff
@@ -14,7 +14,7 @@ class SpamAddress < ApplicationRecord
   validates_uniqueness_of :email, :message => 'This address is already marked as spam'
 
   def self.spam?(email_address)
-    exists?(:email => email_address)
+    Array(email_address).any? { |email| exists?(['email ILIKE ?', email]) }
```

Comment:
Don't think we need this to be this complex.

For starters.. it runs a query for each email address - or at least until a match has been found.
```
pry(main)> SpamAddress.spam? ['foo@bar.com', 'a@localhost']
  SpamAddress Exists (0.4ms)  SELECT  1 AS one FROM "spam_addresses" WHERE (email ILIKE 'foo@bar.com') LIMIT $1  [["LIMIT", 1]]
  SpamAddress Exists (0.4ms)  SELECT  1 AS one FROM "spam_addresses" WHERE (email ILIKE 'a@localhost') LIMIT $1  [["LIMIT", 1]]
```

Where the existing code already handles array of addresses:
```
pry(main)> SpamAddress.exists?(email: ['foo@bar.com', 'a@localhost'])
  SpamAddress Exists (0.6ms)  SELECT  1 AS one FROM "spam_addresses" WHERE "spam_addresses"."email" IN ('foo@bar.com', 'a@localhost') LIMIT $1  [["LIMIT", 1]]
```

So we just need to do a case insensitive search, but we can't do an ILIKE IN query easily (and I think having raw SQL in the source will cause a deprecation warning - see #5568) 

Instead we could switch to the CITEXT column type: https://github.com/rails/rails/commit/0e0b41d58c0a8e5921b7eebc3a6e0e4cc012ce4d